gomock/controller.go

```diff
@@ -70,62 +70,84 @@ type TestReporter interface {
 	Fatalf(format string, args ...interface{})
 }
 
+// TestHelper is a TestReporter that has the Helper method.  It is satisfied
+// by the standard library's *testing.T.
+type TestHelper interface {
+	TestReporter
+	Helper()
+}
+
 // A Controller represents the top-level control of a mock ecosystem.
 // It defines the scope and lifetime of mock objects, as well as their expectations.
 // It is safe to call Controller's methods from multiple goroutines.
 type Controller struct {
 	mu            sync.Mutex
-	t             TestReporter
+	T             TestHelper
```

Comment:
I don't love that T is now a public var. I understand that it needs to be accessed by the mock. It's unfortunate that the testing package API requires direct access to T for T.Helper. I think we should at least add a comment that T is only public so that the mock can use it and any attempt by the user to set it, modify it, call it, etc. is unsupported and could be broken in the future without any consideration for backward compatibility (or something like that).

I wish T.Helper took a number like most other call stack related functions.

Reply:
I will add a comment. I agree it is unfortunate, but I was unable to think of a way around it.